### PR TITLE
feat: Counter component

### PR DESCRIPTION
**Pull request by y-lohse**
A new component to show a simple counter. [Here's a link to the demo and the docs!](https://yannick-lohse.fr/cozy-ui/react/#counter)

There are potentially more things that could be done, like coupling the `max` and the `size` properties, an easier way to change the font color, etc. I don't really know what the needs are at the moment, so I think we can expand it later, but le me know if you think something essential is missing.

There's only one size for now — I think we'll be using the same ones as in #650 once it's settled.

### CODE REVIEW

**Comment by y-lohse** [comment]
[(2) Déplacer plusieurs fichiers/dossiers](https://trello.com/c/zKuIBwvp/1322-2-d%C3%A9placer-plusieurs-fichiers-dossiers)

**Comment by ptbrowne** [comment]
Apart from the color, LGTM.

I think styling should be decoupled from the original component. As we are moving towards themes, I think to support themes, we should have some way of wrapping a component to provide it with `primaryColor`, `secondaryColor` etc...

```jsx
const Counter = props => ...

const counterStyle = theme => ({
  backgroundColor: primaryBackgroundColor,
  color: primaryContrastColor
})
const ThemedCounter = props => (
  <ThemeProvider>{ theme => (
     <Counter {...props} style={counterStyle(theme)} />
  }</ThemeProvider>
)
```

**Comment by y-lohse** [comment]
@ptbrowne Can you look at the latest commit and see if it's what you had in mind?

I agree that tying the palette to the component is not such a good idea after all, it's convenient but might cause problems.

On the other hand I'm less comfortable with a `style` property, especially along other props like `size` and `classNames` which are other ways to influence the look of the component. 
I think component props should be explicit about what they change, so `size` is good for example, and `color` as a way to define the main color of the component also looks good to me. Blanket props like `style` are ok for extreme cases where we need to override everything, but shouldn't be the norm.

Regarding the theming, I think a better approach might be that the component maps the theme colors to it's own props? Something like

```
const Counter =({ color, size, … }) => ...

const ThemedCounter = props => (
  <ThemeProvider>{ theme => (
     <Counter {...props} color={theme.primaryBackgroundColor} />
  }</ThemeProvider>
)
```

Let me know what you think! I'm ok with the PR as it is but I still feel like a `color` prop makes more sense than a generic `style` prop.

**Comment by ptbrowne** [comment]
My suggestion was mainly about not tying palette and naming the property `backgroundColor`, to echo the CSS properties. Otherwise I think we will mix it up. OK for not having styles, it will push for utility classes.

**Comment by ptbrowne** [comment]
Looking twice, I don't think the circle is tied to the Counter. What if we want the `max` (99+) behavior without the circle ?

File: stylus/components/counter.styl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    justify-content center
+    border-radius 50%
+    overflow hidden
+    background-color dodgerBlue
+    color white
+
+$counter--medium
+    width rem(40)
+    height rem(40)
+    font-size rem(20)
```
Comment (ptbrowne) [inline stylus/components/counter.styl]: It seems that for perfect alignment, a negative top margin should be applied.

image image

**Comment by y-lohse** [comment]
> Looking twice, I don't think the circle is tied to the Counter. What if we want the max (99+) behavior without the circle ?

Yeah that's a good point. Do you think it would make sense to have a `Circle` component that would be used in conjunction with the `Counter` and `Avatar` components?

**Comment by ptbrowne** [comment]
Yes I would be for Circle, responsible for size, and Counter inside. It's true that Avatar could be done with this but I would not do that now.

**Comment by y-lohse** [comment]
Ok done! The counter and the circle are now separate components.

**Pull request merged**